app/models/parameter.rb

```diff
@@ -83,7 +83,7 @@ def set_default_key_type
   end
 
   def set_searchable_value
-    self.searchable_value = Parameter.format_value_before_type_cast(value, key_type)
+    self.searchable_value = key_type == 'boolean' ? value.to_s : Parameter.format_value_before_type_cast(value, key_type)
```

Comment:
Just a question, why is the condition here and not in `Parameter.format_value_before_type_cast`? It seems we handle other types there already.

Reply:
format_value_before_type_cast is used at multiple places. I wasn't sure if boolean type is supposed to be returned as a string for every place that it's called.

Reply:
I'd say try to make the list of where it's used and consider the impact for each such item. If we don't find a place where this would cause a problem, we should keep the logic at one place. Otherwise in future, when someones modifies the behavior of `Parameter.format_value_before_type_cast` there's a risk they will miss this place and it will remain not updated.

Reply:
@ares, I think the problem is a bit different. I have tried to write it in the [comment](https://github.com/theforeman/foreman/pull/9966#issuecomment-1873305248) on this PR. 
TL;DR: we don't know the parameter type in the search function. I think we should solve it from there, it's not a problem with value serialization.

Reply:
I agree, this solution would help only to some degree. If we'd go with this solution, I'd like to see the serializing logic on one place and instead of staying on the safe side, we should go case by case and see what the impact would be. But if we're not going this route and the resulting patch will be different, of course this is no longer relevant. The main principle remains valid (don't split the code just due to uncertainty of impact)